src/Plugin/Commerce/PaymentGateway/PaypalPaymentGatewayTrait.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\commerce_paypal\Plugin\Commerce\PaymentGateway;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Provides common methods to be used by PayPal payment gateways.
+ */
+trait PaypalPaymentGatewayTrait {
+
+  /**
+   * Loads the payment for a given remote id.
+   *
+   * @param string $remote_id
+   *   The remote id property for a payment.
+   *
+   * @return \Drupal\commerce_payment\Entity\PaymentInterface
+   *   Payment object.
+   */
+  public function loadPaymentByRemoteId($remote_id) {
+    /** @var \Drupal\commerce_payment\PaymentStorage $storage */
+    $storage = \Drupal::service('entity_type.manager')->getStorage('commerce_payment');
+    $payment_by_remote_id = $storage->loadByProperties(['remote_id' => $remote_id]);
+    return reset($payment_by_remote_id);
+  }
+
+  /**
+   * Processes an incoming IPN request.
+   *
+   * @param Request $request
+   *   The request.
+   *
+   * @return mixed
+   *   The request data array or FALSE.
+   */
+  public function processIpnRequest(Request $request) {
+    // Validate and get IPN request data.
+    $ipn_data = $this->getIpnRequestValidate($request);
```

Comment:
Since this method is pretty empty, let's just keep validation in here as part of the process. We can always split out later if needed. Otherwise we'd just need to call "validate".